### PR TITLE
2095 upgrade django

### PR DESCRIPTION
**Pull request by adrian-lara**
**NOTE:** This PR contains dependency updates AS WELL AS django migrations forced by the Django upgrade. See collected migration notes below.

#### What's this PR do?
Upgrade Django and other API-related dependencies.

#### How should this be manually tested?
Monkey test the app.

In order to review the actual code changes, I've grouped and organized the changes using commits. 
*None of the commits "overlap", so a reasonable approach for reviewing could be to review one commit at a time. The commit message and extra commit details provide background on each change.

*This is true for most commits except for the merge commits and the last few other commits, which fix bugs I found during my own monkey testing.

#### What are the relevant tickets?
#2095 

#### Migration Notes
- admin.LogEntry -related migration. I couldn't find release notes on this but here's the PR. https://github.com/django/django/commit/c4158d050f43bb65aad224f4cf7ae56d9e630445
- Group.name max_length is increased from 80 to 150 characters. (https://docs.djangoproject.com/en/3.0/releases/2.2/#miscellaneous)
- https://docs.djangoproject.com/en/3.0/releases/2.2/#permissions-for-proxy-models

### CODE REVIEW

**Comment by adrian-lara** [comment]
@nllong and/or @axelstudios tagged you both a few hours early, but this is ready for review now 😅

**Comment by nllong** [comment]
Thanks for testing everyone! Any reservations on merging this today?

File: seed/api/base/views.py [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
 from seed.decorators import ajax_request
-from rest_framework.decorators import list_route
+from rest_framework.decorators import action
 
 
 class TestReverseViewSet(viewsets.ViewSet):
     raise_exception = True
 
-    @list_route(methods=['GET'])
+    @action(detail=False, methods=['GET'])
```
Comment (nllong) [inline seed/api/base/views.py]: this will take a while to get used to...

File: seed/tasks.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def invite_to_seed(domain, email_address, token, user_pk, first_name):
     Returns: nothing
     """
     signup_url = reverse_lazy('landing:signup', kwargs={
-        'uidb64': urlsafe_base64_encode(force_bytes(user_pk)).decode(),
+        'uidb64': urlsafe_base64_encode(force_bytes(user_pk)),
```
Comment (nllong) [inline seed/tasks.py]: so no more encoding to decode

File: seed/tests/test_properties_serializers.py [hunk 3]
```diff
@@ -358,11 +358,12 @@ def test_update_put(self, mock_serializer):
         mock_serializer.return_value.is_valid.return_value = True
         mock_serializer.return_value.save.return_value = self.property_state
         mock_request = mock.MagicMock()
+        property = FakePropertyFactory(organization=self.org).get_property()
```
Comment (nllong) [inline seed/tests/test_properties_serializers.py]: 👍

**Pull request merged**